feat: add docker proxy

egoarka opened the pull request:
смотреть readme.md

egoarka committed:
feat: add docker proxy

egoarka committed:
chore: rename proxy

andrsuh commented:
Эта прокся не помешает запускать наши сервисы в сворме?

andrsuh commented:
В чем вообще бенефит?

egoarka commented:
она понадобится только для учеток, которые не имеют админские права


бенифиты в ридми

andrsuh commented:
В ридми мало инфы

Получается я просто добавляю то, что хочу запуллить в allowed-images.txt и все

andrsuh commented on services/docker-proxy/docker-proxy.py:
```diff
@@ -0,0 +1,86 @@
+
+docker_admin = "/tmp/docker-proxy.sock"
+
+if os.path.exists(docker_admin):
```
обясни плз

andrsuh replied:
мы не хотим, чтобы к процессу докеры ходили напрямую, так?

egoarka replied:
это не процесс докера, это юникс сокет, к нему доступ должна иметь только админская учетка

andrsuh replied:
мне не надо сюда писать) я понял, что у тебя написано) Сделай в ридми подробнее и комменты к коду, пожалуйста)

egoarka commented:
неа, allowed images в коде не используются, я их добавил на случай если понадобятся ограничения на конкретные образы

прокси поднимается на админской учетке

а другие учетки используют в качестве основного сокета для клиента докера

egoarka commented:
>Эта прокся не помешает запускать наши сервисы в сворме?

нет, эта прокся будет использоваться только на обычных учетках ( не рутовых )

andrsuh commented on services/docker-proxy/readme.md:
```diff
@@ -0,0 +1,29 @@
+
+## использование
+
+`export DOCKER_HOST="unix:///tmp/docker-proxy.sock"`
```
Объясни подробнее, что мы перенаправляем команды докеры к твоему коду

egoarka replied:
да, так и есть

docker cli формирует на своей стороне запрос с командой, формирует его и отправляет по сокету заданному в переменной окружении DOCKER_HOST

andrsuh commented on services/docker-proxy/docker-proxy.py:
```diff
@@ -0,0 +1,86 @@
+]
+
+
+def listenToClient(conn, addr):
```
добавь комментарии, что то, что приходит в твой сокет и проксируешь в сокет докера, при этом проверяя, дозволено или нет :)

egoarka replied:
добавил

andrsuh commented on services/docker-proxy/docker-proxy.py:
```diff
@@ -0,0 +1,86 @@
+
+    # print(endpoint)
+
+    if any(item in endpoint for item in create_endpoints):
```
может, было проще перечислить то, что можно, а не то что нельзя?

egoarka replied:
для первой версии не ставил такой задачи перед собой

если понадобится, можно будет обновить код под наши требования

andrsuh commented:
Спасибо, хорошая идея! И круто, что на питоне

egoarka committed:
chore: refresh docs

andrsuh commented:
не забывай еще и про трелло, чтобы я понимал кто это делал и тд. @valtornist99 , это для тебя тоже) и я хочу более намного более подробный ридми и комменты к коду)

egoarka committed:
chore: add recv_timeout
feat: add image exists (+ hander)
feat: add allowed command (+ handler)
chore: logging for incoming and outcoming

egoarka committed:
chore: clarify allowed command

egoarka committed:
feat: add service create handler

egoarka commented:
добавил /services/create эндпоинт

для stack deploy'я он используется в том числе
